src/components/RequestSignModal.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  Button,
+  Center,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalOverlay,
+  type ModalProps,
+} from '@chakra-ui/react';
+import { Trans } from '@lingui/macro';
+import type { FunctionComponent } from 'react';
+
+import { SignIcon } from './icons';
+
+export type RequestSignModalProps = ModalProps & {
+  mode: 'positive' | 'negative';
+  headerIcon: JSX.Element;
+  buttonText: string;
+  isLoading: boolean;
+  buttonDisabled?: boolean;
+  onSignButtonClick: () => void;
+};
+
+/**
+ * Render a modal with header, body and button to request signature.
+ *
+ * @param props - The component props.
+ * @param props.isOpen - Whether the modal is open.
+ * @param props.onClose - A function to close the modal.
+ * @param props.mode - The mode of the modal.
+ * @param props.headerIcon - The icon to render as a header of the modal.
+ * @param props.children - The children to render inside the modal.
+ * @param props.buttonText - The text to be shown on sign button.
+ * @param props.onSignButtonClick - A function to be called when Sign button is clicked.
+ * @param props.isLoading - Whether the modal is loading.
+ * @param props.buttonDisabled - Whether the button is disabled.
+ * @returns A React component.
+ */
+export const RequestSignModal: FunctionComponent<RequestSignModalProps> = ({
+  isOpen,
+  onClose,
+  mode = 'positive',
```

Comment:
not a must, but it looks like the mode is just for handle the button colour?  why not just let the component pass in the button bg color?

Reply:
I do get the point. I kept it to set a mode of modal, so it can be used for other styling too in future. As currently the icons are separate for red and blue, I am expecting one icon for both and configure using mode in future.

Reply:
good to know that, resolve